gdk4/src/texture.rs

```diff
@@ -15,4 +20,16 @@ impl<O: IsA<Texture>> TextureExtManual for O {
             ffi::gdk_texture_download(self.as_ref().to_glib_none().0, data.as_mut_ptr(), stride);
         }
     }
+
+    #[cfg(any(feature = "v4_6", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v4_6")))]
+    fn download_float(&self, data: &mut [f32], stride: usize) {
+        unsafe {
+            ffi::gdk_texture_download_float(
+                self.as_ref().to_glib_none().0,
+                data.as_mut_ptr(),
+                stride,
```

Comment:
I will drop this one from manual bindings, Company wanted to remove it before 1.50 release and that will probably happen in 1.51.

Reply:
I'm not sure though what kind of check I need to do for the slice, it depends on the format of the texture at the end; isn't it?

Reply:
Assuming ARGB/BGRA/etc, the stride must be `>= 4 * width`, and the whole slice must be `== 4 * stride * height`. So yes, you need to check the format and width and height.

Reply:
Except the texture doesn't give us any information about it's internal format. It can be a MemoryTexture/GLTexture as well so it's probably not an easy way to validate this one

Reply:
Well, read the docs (for `gdk_texture_download()`) :)

```
The data format of the downloaded data is equivalent to CAIRO_FORMAT_ARGB32, so every downloaded pixel requires 4 bytes of memory.
```

Reply:
Thanks! I opened https://github.com/gtk-rs/gtk4-rs/issues/677 to track it, I would leave it for a different PR as it's not related to the gtk 4.6 APIs